### PR TITLE
Nataliia

### PR DESCRIPTION
**Pull request by natalyshevchuk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 .venv/
-.pytest_cache/
\ No newline at end of file
+.pytest_cache/
+.idea/
+reports/
```
Comment (TanyaKalistruk) [inline .gitignore:4]: the empty line at the end of file

File: application.py [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
     def __init__(self):
         self.session = requests.Session()
 
     def authentication(self, username, password):
-        self.request = self.session.post(LOGIN_URL, auth=(username, password))
-        if self.request.status_code == 200:
+        request = self.session.post(LOGIN_URL, auth=(username, password))
+        if request.status_code == 200:
             self.get(BASE_URL)
             self.session.headers.update({"X-XSRF-TOKEN": self.session.cookies.get_dict()['XSRF-TOKEN'],
```
Comment (TanyaKalistruk) [inline application.py:20]: how about making "X-XSRF-TOKEN" and "XSRF-TOKEN" as constants?

File: application.py [hunk 3]
```diff
@@ -38,7 +41,6 @@ def delete(self, endpoint, data='', headers=HEADERS):
         self.request = self.session.delete(endpoint, data=converted_data, headers=headers)
         return self.request
 
-    def debug_info(self):
-        print(self.request.status_code, '\n', self.request.text)
-        from pprint import pprint
-        pprint(self.request.json())
+    def check_200(self):
```
Comment (TanyaKalistruk) [inline application.py:44]: why not do this assertion in tests?

File: data_tests/auth.py [hunk 4]
```diff
@@ -1,9 +1,4 @@
-from base64 import b64decode as decode
-
-TOKEN = "3e83667c-c59c-4fda-aa7a-a47346a3cd6a"
-
-EMAIL_FORGOT_PASSWORD = "rabotynet.test.fp@gmail.com"
-FROM_FORGOT_PASSWORD = decode(b'cm9tYV9leHBlcnQ=').decode()
+TOKEN = {'token': "3e83667c-c59c-4fda-aa7a-a47346a3cd6a"}
```
Comment (TanyaKalistruk) [inline data_tests/auth.py:1]: what is this token for?

File: tests/test_auth.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+""" Test case for sign in and sign up """
+import pytest
+import allure
+from data_tests.auth import REGISGER_DATA, TOKEN
+from utilities.db import change_varification_link
+from base import LOGIN_URL, LOGOUT_URL, USER_CONFIRM_EMAIL_URL, USER_REGISTER_URL
+
+
+@allure.story('Sign In')
+@pytest.mark.parametrize('user, password, expected', [
```
Comment (TanyaKalistruk) [inline tests/test_auth.py:10]: username, password, expected_role?

File: tests/test_auth.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+def test_login(app, user, password, expected):
+    """ Test for authenticate and authorize user.
+    doesn't use application method, to check auth step by step"""
+    session = app.session
+    response = session.post(LOGIN_URL, auth=(user, password))
+    with allure.step("Check if status code equal 200"):
+        assert response.status_code == 200, "Wrong status code"
+    data = response.json()
+    with allure.step("Check if username is correct"):
+        assert data['username'] == user, "Wrong username"
```
Comment (TanyaKalistruk) [inline tests/test_auth.py:24]: how about making a separate file with all names of fields and reuse it there?
USERNAME = 'username'
LOGIN = 'login'
etc.

File: tests/test_auth.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    with allure.step("Check if user authorized correctly"):
+        assert data['authorities'][0]['authority'] == expected, "Wrong user role"
+    session.get(LOGOUT_URL)
+
+
+@allure.feature('Sign Up')
+def test_sign_up_begin(app):
+    """ Start to register user. Up to send email """
+    app.post(USER_REGISTER_URL, data=REGISGER_DATA)
+    app.check_200()
```
Comment (TanyaKalistruk) [inline tests/test_auth.py:34]: why is there used a method for checking status_code, but before this assertion was in the test?

File: tests/test_auth.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    with allure.step("Check if username is correct"):
+        assert data_received['login'] == REGISGER_DATA['login'], "Wrong username"
+
+
+@allure.feature('Sign Up')
+def test_sign_up_end(app):
+    """ Finish to register user. From receive email"""
+    change_varification_link(REGISGER_DATA['login'])
+    app.get(USER_CONFIRM_EMAIL_URL, data=TOKEN)
+    app.check_200()
```
Comment (TanyaKalistruk) [inline tests/test_auth.py:45]: The same. The assertion will be better to do in tests

File: tests/test_delete_company.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import allure
+from base import COMPANY_EXISTS_URL, DELETING_OF_COMPANY
+
+
+@allure.feature('Deleting a company')
+def test_delete_company(app):
+    """Deleting of the company"""
+
+    with allure.step("Login"):
+        app.authentication("cowner@gmail.com", "cowner")
```
Comment (TanyaKalistruk) [inline tests/test_delete_company.py:11]: why username and password are there like strings?

File: tests/test_update_company_name.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+from data_tests.cowner_data import UPDATE_COMPANY_NAME_DATA, UPDATED_NAME
+from base import UPDATE_COMPANY_URL
+
+
+@allure.feature('Updating of the company name')
+def test_update_company_name(app):
+    """Updating of the company name"""
+
+    with allure.step("Login"):
+        app.authentication("cowner@gmail.com", "cowner")
```
Comment (TanyaKalistruk) [inline tests/test_update_company_name.py:12]: the same with creds

Comment (natalyshevchuk) [reply tests/test_update_company_name.py:12]: Done

File: tests/test_delete_company.py (hunk 9 above)
Comment (natalyshevchuk) [reply tests/test_delete_company.py:11]: I have changed it

File: application.py (hunk 2 above)
Comment (Vilkoc) [reply application.py:20]: they change each time when tests start.

File: application.py (hunk 3 above)
Comment (Vilkoc) [reply application.py:44]: to avoid code repeating

File: data_tests/auth.py (hunk 4 above)
Comment (Vilkoc) [reply data_tests/auth.py:1]: this is mock for verification_token for register new user and recover password. Some times it takes about 2 minutes to get email with verification link. To skip this i put token into DB and  use it as verification token

File: tests/test_auth.py (hunk 5 above)
Comment (Vilkoc) [reply tests/test_auth.py:10]: fixed

File: tests/test_auth.py (hunk 7 above)
Comment (Vilkoc) [reply tests/test_auth.py:34]: fixed

**Pull request closed without merging**